src/dropdown.ts

```diff
@@ -2507,24 +2532,34 @@ export default class Dropdown extends LitElement implements FormControl {
     } else if (
       this.multiple &&
       event.target instanceof DropdownOption &&
-      event.target.selected &&
-      event.target.value
+      event.target.selected
     ) {
-      this.#value.push(event.target.value);
+      if (event.target.value) {
+        this.#value.push(event.target.value);
+      }
+
+      this.selectedAndEnabledOptions = [
+        ...this.selectedAndEnabledOptions,
+        event.target,
+      ];
+
+      this.#setTagOverflowLimit();
     } else if (
       event.target instanceof DropdownOption &&
       event.target.selected &&
-      // We're only concerned with an enabled option that's the last selected option
-      // because we're single-select, and only the last selected option affects
-      // Dropdown's `value` and and its input field.
-      event.target === this.lastSelectedAndEnabledOption &&
```

Comment:
Removed this condition so I could update `this.selectedAndEnabledOptions` even when the selected option that was just programmatically enabled isn't the last selected option. Also, as a result, had to tweak setting `#value` below.